e2e/test_module.py

```diff
@@ -53,18 +52,29 @@ def prepare_tests():
     tests = []
 
     test_path = Path().cwd()
+
     for module_test_dir in test_path.iterdir():
         if not module_test_dir.is_dir() or not module_test_dir.name.endswith(
             TestConstants.TEST_MODULE_DIR_SUFFIX
         ):
             continue
 
-        for test_dir in module_test_dir.iterdir():
-            if not test_dir.is_dir():
+        for test_or_group_dir in module_test_dir.iterdir():
+            if not test_or_group_dir.is_dir():
```

Comment:
Why are we here skipping something that is not `dir`?

Reply:
The current E2E test organization is `e2e/ALGORITHM_NAME/TEST_NAME/{test_files}`. The line in question checks if the contents of the `ALGORITHM_NAME` dir are directories (since they contain the test files) and skips the rest.